soroban-env-common/src/env.rs

```diff
@@ -168,7 +168,7 @@ pub trait EnvBase: Sized + Clone {
     fn string_new_from_slice(&self, slice: &[u8]) -> Result<StringObject, Self::Error>;
 
     /// Form a new `Symbol` host object from a slice of client memory.
-    fn symbol_new_from_slice(&self, slice: &str) -> Result<SymbolObject, Self::Error>;
+    fn symbol_new_from_slice(&self, slice: &[u8]) -> Result<SymbolObject, Self::Error>;
```

Comment:
If this change requires rewriting any macros upstream, I think we should add a u8 variant alongside the str variant rather than replacing. We're not removing str from the other fns like `map_new_from_slices`, so removing the str variant of symbol_new_ doesn't seem to achieve much from a holistic pov of this interface.

Reply:
It doesn't require any macro changes, just one respective change in the SDK implementation of `EnvBase`. SDK itself just uses `Symbol::try_from_val` in its wrapper.
I'm not strictly opposed to keeping 2 versions, but since one of them is in fact redundant, I would like to get rid of unnecessary maintenance burden and keep just one version. Again, if we're worried about hypothetical alternative SDK that uses this, I would rather only keep the old version (again, maintaining 2 versions seems unnecessary and will cause more confusion in the future).